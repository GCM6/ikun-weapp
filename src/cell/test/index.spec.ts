import path from 'node:path';
import simulate from 'miniprogram-simulate';

describe('cell', () => {
  const KCell = simulate.load(path.resolve(__dirname, '../cell'), 'k-button', {
    rootPath: path.resolve(__dirname, '../../'),
  });

  test('should render cell and match snapshot', () => {
    const comp = simulate.render(KCell);
    comp.attach(document.createElement('parent-wrapper'));

    expect(comp.toJSON()).toMatchSnapshot();
  });

  test('should emit click event', async () => {
    const comp = simulate.render(
      simulate.load({
        usingComponents: {
          'k-cell': KCell,
        },
        template: `<k-cell id="comp" bind:click="onClick" />`,
        data: {
          value: 0,
        },
        methods: {
          onClick() {
            this.setData({ value: this.data.value + 1 });
          },
        },
      })
    );
    comp.attach(document.createElement('parent-wrapper'));

    const button = comp.querySelector('#comp >>> .k-cell');

    button?.dispatchEvent('tap');
    await simulate.sleep(10);
    expect(comp.data.value).toEqual(1);
  });
});
