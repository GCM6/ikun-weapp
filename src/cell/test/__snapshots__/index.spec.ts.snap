// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cell should render cell and match snapshot 1`] = `
<wx-view
  ariaLabel=""
  ariaRole=""
  class="k-class k-cell"
  hoverClass="k-cell--hover "
  hoverStayTime="70"
  id=""
  style=""
  bind:tap="onTap"
>
  <wx-view
    class="k-cell__title title-class"
  >
    <wx-view
      class="k-cell__label label-class"
    >
      
    </wx-view>
  </wx-view>
  <wx-view
    class="k-cell__value value-class"
  />
</wx-view>
`;
